modules/jenkins/manifests/init.pp

```diff
@@ -1,5 +1,36 @@
+#init.pp for adding jenkins with latest version 
+
+#Class that rolls it all together
 class jenkins {
   package { 'jenkins':
-    ensure => installed,
+    ensure => latest, 
+    require => [ 
+                 File['/etc/apt/sources.list.d/jenkins.list'],
+                 Exec['add_jenkins_signature'],
+                 Exec['jenkins_apt_get_update']
+               ]
+        
   }
+ 
+  #create apt source list file for jenkins 
+  file { '/etc/apt/sources.list.d/jenkins.list':
+       ensure => file, 
+       mode   => '0644',
+       owner  => 'root',
+       group  => 'root',
+       source => 'puppet:///modules/jenkins/etc/apt/sources.list.d/jenkins.list', 
+   }
+
+
+  # Execute line for Adding Jenkins gpg signature to host
+  exec {'add_jenkins_signature':
+           command => "/usr/bin/wget -q -O - http://pkg.jenkins-ci.org/debian/jenkins-ci.org.key | /usr/bin/apt-key add -"
```

Comment:
Sure. Could I get some help on writing the line? Not sure exactly how to go about doing that aside from possibly adding another file with the key directly.